src/ethereum/ethNetwork.js

```diff
@@ -1487,6 +1537,154 @@ export class EthereumNetwork {
     })
   }
 
+  async checkTxsBlockbook(
+    startBlock: number = 0
+  ): Promise<EthereumNetworkUpdate> {
+    const address = this.ethEngine.walletLocalData.publicKey.toLowerCase()
+    let page = 1
+    let totalPages = 1
+    const out = {
+      newNonce: '0',
+      tokenBal: {},
+      tokenTxs: {},
+      server: ''
+    }
+    while (page <= totalPages) {
+      const query =
+        '/api/v2/address/' +
+        address +
+        `?from=${startBlock}&page=${page}&details=txs`
+      const { result: jsonObj, server } = await this.multicastServers(
+        'blockbookTxs',
+        query
+      )
+      let addressInfo
+      try {
+        addressInfo = asBlockbookAddress(jsonObj)
+      } catch (e) {
+        this.ethEngine.log(`checkTxsBlockbook ${server} ${e}`)
+        throw new Error(`Blockbook ${server} returned invalid JSON`)
+      }
+      const { nonce, tokens, balance, transactions } = addressInfo
+      out.newNonce = nonce
+      out.tokenBal.ETH = balance
+      out.server = server
+      totalPages = addressInfo.totalPages
+      page++
+
+      // Token balances
+      for (const token in tokens) {
+        const { symbol, balance } = tokens[token]
+        out.tokenBal[symbol] = balance
+      }
+
+      // Transactions
+      for (const tx of transactions) {
+        const transactionsArray = []
+        try {
+          const cleanTx = asBlockbookTx(tx)
+          if (
+            cleanTx.tokenTransfers !== undefined &&
+            cleanTx.tokenTransfers.length > 0
+          ) {
+            for (const tokenTransfer of cleanTx.tokenTransfers) {
+              if (
+                address === tokenTransfer.to.toLowerCase() ||
+                address === tokenTransfer.from.toLowerCase()
+              ) {
+                transactionsArray.push(
+                  this.processBlockbookTx(tx, tokenTransfer)
+                )
+              }
+            }
+          }
+          if (
+            address === tx.vout[0].addresses[0].toLowerCase() ||
+            address === tx.vin[0].addresses[0].toLowerCase()
+          )
+            transactionsArray.push(this.processBlockbookTx(tx))
+        } catch (e) {
+          continue
```

Comment:
If anything throws you should stop processing completely. Otherwise you may advance the block height and never query the problematic range